gems/sorbet-runtime/test/types/types.rb

```diff
@@ -442,6 +442,24 @@ def subtype_of?(other)
         allocs_when_invalid = counting_allocations {type.valid?(invalid)}
         assert_equal(0, allocs_when_invalid)
       end
+
+      it 'gives the right error when passed an unrelated enumerable' do
+        type = T::Array[Integer]
+        Object.const_set(:Foo, Class.new do
+          include Enumerable
+
+          def each
+            yield "something"
+          end
+        end)
+        value = Foo.new
+
+        msg = type.error_message_for_obj(value)
+        expected_error = "Expected type T::Array[Integer], got Foo"
+        assert_equal(expected_error, msg)
+      ensure
+        Object.send(:remove_const, :Foo)
+      end
```

Comment:
I don't like the fact that I am doing this, but I have to give a name to the class for the test to run properly, and I couldn't think of a better way of doing it. I am open to other ideas.

Reply:
Could you define the class as a proper named class at the top of the file, sort of like a fixture?

We make a lot of helper classes in other files that way:

https://github.com/sorbet/sorbet/blob/0c301f6c2900528f9bd9a7dd869f574d4504c644/gems/sorbet-runtime/test/types/props/struct.rb#L5-L14

^ e.g., these structs are each used only in one `it` block.

Reply:
Ah, makes much more sense. I'll change it the class to a fixture like those.